base/linked_list.jl

```diff
@@ -1,7 +1,15 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
+
+"""
+    Base.IntrusiveLinkedList{T}
+
+Is used like this Base.IntrusiveLinkedList{Base.LinkedListItem{T}} as LinkedList{T}.
+Holds the first (head) and the last (tail) `Base.LinkedListItem{T}` in the list.
```

Comment:
But which list? This text seems to be written with a specific scenario in mind; any reader who doesn't have that will have a hard time following.

I think it is better to assume less and write something along these lines:

> This type implements the anchover for "intrusive" linked list. It stores
> references to the first (head) and the last (tail) entry of the list.
> List entries are of type `Base.LinkedListItem{T}`.

If you want to explain usage, I'd suggest to include a complete usage example,
perhaps even insider a `jldoctest` block?

Finally, I guess a `Base.LinkedListItem` docstring would also be helpful